twake/backend/node/src/services/previews/services/engine/service.ts

```diff
@@ -0,0 +1,95 @@
+import { PreviewPubsubHandler, PreviewServiceAPI } from "../../api";
+import { logger, TwakeContext } from "../../../../core/platform/framework";
+import _ from "lodash";
+import { PubsubServiceAPI } from "../../../../core/platform/services/pubsub/api";
+import { PreviewPubsubCallback, PreviewPubsubRequest, ThumbnailResult } from "../../types";
+import { getTmpFile } from "../../utils";
+import fs from "fs";
+import { unlink } from "fs/promises";
+import StorageAPI from "../../../../core/platform/services/storage/provider";
+
+/**
+ * Push new message notification to a set of users
+ */
+export class PreviewProcessor
+  implements PreviewPubsubHandler<PreviewPubsubRequest, PreviewPubsubCallback> {
+  constructor(
+    service: PreviewServiceAPI,
+    private pubsub: PubsubServiceAPI,
+    readonly storage: StorageAPI,
+  ) {
+    this.service = service;
+  }
+  service: PreviewServiceAPI;
+  init?(context?: TwakeContext): Promise<this> {
+    throw new Error("Method not implemented.");
+  }
+
+  readonly topics = {
+    in: "services:preview",
+    out: "services:preview:callback",
+  };
+
+  readonly options = {
+    unique: true,
+    ack: true,
+  };
+
+  name = "PreviewProcessor";
+
+  validate(message: PreviewPubsubRequest): boolean {
+    return !!true;
```

Comment:
We could improve the validation here, check if the "message" contain a document etc